.github/workflows/v8upgrade.yml

```
@@ -0,0 +1,52 @@
+name: V8 Upgrade
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # Run every day
+
+jobs:
+    upgrade:
+        name: Upgrade V8
+        runs-on: ubuntu-18.04
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+              with:
+                  submodules: true
+                  fetch-depth: 0
+            - name: Check latest v8 stable version
+              run: cd deps && ./check_latest_v8_version.py
+            - name: Update depot_tools fetch config
+              run: cd deps/depot_tools && git config --unset-all remote.origin.fetch; git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/*
+              shell: bash
+            - name: Add depot_tools to PATH
+              run: echo "$PWD/deps/depot_tools" >> $GITHUB_PATH
+              shell: bash
+            - name: Fetch V8
+              run: cd deps && fetch v8 || true
+            - name: Get latest stable commit ref of v8
+              id: v8_latest_commit
+              run: echo ::set-output name=VALUE::$(curl -sS 'https://omahaproxy.appspot.com/all.json?os=linux&channel=stable' | jq '.[0]["versions"][0]["v8_commit"]')
+            - name: Get latest stable version number of v8
+              id: v8_latest_version
+              run: echo ::set-output name=VALUE::$(curl -sS 'https://omahaproxy.appspot.com/all.json?os=linux&channel=stable' | jq '.[0]["versions"][0]["v8_version"]')
+            - name: Checkout v8 to stable commit ref
+              run: cd deps/v8 && git fetch && git checkout ${{steps.v8_latest_commit.outputs.VALUE}}
```

Looks like we could just checkout the version number tag, to avoid needing to also get the commit SHA.  As in, it looks like this would work

```suggestion
              run: cd deps/v8 && git fetch && git checkout ${{steps.v8_latest_version.outputs.VALUE}}
```

---

Is there actually a branch/tag for the full version? I thought that it was either checking out the v8 commit specifically or parsing the full version for the major+minor to compose the specific branch name: #8 in https://github.com/rogchap/v8go#upgrading-the-v8-binaries

---

Yes, there is a tag.  For instance, this PRs deps/v8_version is 9.0.257.18 and there is a tag (https://chromium.googlesource.com/v8/v8.git/+/refs/tags/9.0.257.18).  If you `cd deps/v8 && git checkout 9.0.257.18` then it checks out that commit.

> Find the right branch-heads/** to checkout, for example if the v8_version is 8.7.220.31 then you want to git checkout branch-heads/8.7. You can check all the branch-heads with git branch --remotes | grep branch-heads/

It seems like the upgrade instructions would have led to not using the stable release version that it recommends looking up by checking out a branch instead.

---

FYI: the branch naming upgrade instructions came from the official docs: https://v8.dev/docs/release-process#stable-releases

---

Oh, I guess they don't recommend the version chrome uses, since they aren't going to release fixes as frequently for a client application where the end user isn't going to want to upgrade their application that frequently.  So using the branch head might reduce the time to get security patches ready and released.